src/position-manager/interfaces/IAllowancePositionManager.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity ^0.8.0;
+
+import {EIP712Types} from 'src/libraries/types/EIP712Types.sol';
+import {IPositionManagerBase} from 'src/position-manager/interfaces/IPositionManagerBase.sol';
+
+/// @title IAllowancePositionManager
+/// @author Aave Labs
+/// @notice Interface for position manager handling withdraw permit & credit delegation actions.
+interface IAllowancePositionManager is IPositionManagerBase {
+  /// @notice Thrown when the withdraw allowance is insufficient.
+  error InsufficientWithdrawAllowance(uint256 allowance, uint256 required);
+  /// @notice Thrown when the credit delegation allowance is insufficient.
+  error InsufficientCreditDelegation(uint256 allowance, uint256 required);
+
+  /// @notice Emitted when `owner` approves `spender` to withdraw `amount` for `reserveId` on their behalf.
+  /// @param owner The address of the owner.
+  /// @param spender The address of the spender.
+  /// @param reserveId The identifier of the reserve on the connected `spoke`.
+  /// @param amount The amount of allowance.
+  event WithdrawApproval(
+    address indexed owner,
+    address indexed spender,
+    uint256 indexed reserveId,
+    uint256 amount
+  );
+
+  /// @notice Emitted when `owner` approves `spender` to borrow `amount` from `reserveId` on their behalf.
+  /// @param owner The address of the owner.
+  /// @param spender The address of the spender.
+  /// @param reserveId The identifier of the reserve on the connected `spoke`.
+  /// @param amount The amount of credit delegation.
+  event CreditDelegation(
+    address indexed owner,
+    address indexed spender,
+    uint256 indexed reserveId,
+    uint256 amount
+  );
+
+  /// @notice Approves a spender to withdraw assets from the specified reserve on the connected `spoke`.
+  /// @param spender The address of the spender to receive the allowance.
+  /// @param reserveId The identifier of the reserve.
+  /// @param amount The amount of allowance.
+  function approveWithdraw(address spender, uint256 reserveId, uint256 amount) external;
+
+  /// @notice Approves a spender to withdraw assets from the specified reserve on the connected `spoke` via signature.
+  /// @param params The structured WithdrawPermit parameters.
+  /// @param signature The signed bytes for the intent.
+  function approveWithdrawWithSig(
+    EIP712Types.WithdrawPermit calldata params,
+    bytes calldata signature
+  ) external;
+
+  /// @notice Approves a credit delegation allowance for a spender.
+  /// @param spender The address of the spender to receive the allowance.
+  /// @param reserveId The identifier of the reserve.
+  /// @param amount The amount of allowance.
+  function delegateCredit(address spender, uint256 reserveId, uint256 amount) external;
+
+  /// @notice Approves a credit delegation allowance for a spender via signature.
```

Comment:
worth mentioning it's EIP712 complaint.

Reply:
2e28199f55c43830e8a3841f17f14be1a1bdc76b